eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1468,6 +1469,74 @@ public function publishVersion(APIVersionInfo $versionInfo, array $translations
         return $content;
     }
 
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    protected function copyNonTranslateableFieldsFromMainPublishedVersion(APIContent $content): void
+    {
+        $versionInfo = $content->getVersionInfo();
+        $contentType = $content->getContentType();
+        $initialLanguageCode = $versionInfo->getContentInfo()->getMainLanguageCode();
+        $currentVersionLanguageCode = $versionInfo->initialLanguageCode;
+
+        if ($initialLanguageCode !== $currentVersionLanguageCode) {
+            return;
+        }
+
+        $versions = $this->loadVersions($versionInfo->getContentInfo(), APIVersionInfo::STATUS_DRAFT);
+        $versions = array_filter($versions, static function ($versionInfo) use ($initialLanguageCode) {
+            return $versionInfo->initialLanguageCode !== $initialLanguageCode;
+        });
+
+        $fieldValues = [];
+        $spiFields = [];
+        foreach ($content->getFields() as $field) {
+            $fieldDefinition = $contentType->getFieldDefinition($field->fieldDefIdentifier);
+            $fieldValues[$fieldDefinition->identifier][$field->languageCode] = $field->getValue();
+
+            if ($fieldDefinition->isTranslatable) {
+                continue;
+            }
+
+            $fieldType = $this->fieldTypeRegistry->getFieldType(
+                $fieldDefinition->fieldTypeIdentifier
+            );
+
+            $spiFields[] = new SPIField(
+                [
+                    'id' => $content->getField($fieldDefinition->identifier, $field->languageCode)->id,
+                    'fieldDefinitionId' => $fieldDefinition->id,
+                    'type' => $fieldDefinition->fieldTypeIdentifier,
+                    'value' => $fieldType->toPersistenceValue($field->getValue()),
+                    'languageCode' => $field->languageCode,
+                    'versionNo' => $versionInfo->versionNo,
+                ]
+            );
+        }
+
+        foreach ($versions as $versionInfo) {
+            $updateStruct = new SPIContentUpdateStruct();
+            $updateStruct->name = $this->nameSchemaService->resolveNameSchema(
+                $content,
+                $fieldValues,
+                $versionInfo->languageCodes,
+                $contentType
+            );
+            $updateStruct->fields = $spiFields;
```

Comment:
You're basically overwriting all fields. I'm not sure if as a draft owner I'd like that.

Reply:
It only overrides non-translateable fields.